testing/kokoro/system_tests.sh

```diff
@@ -194,8 +204,16 @@ date
 
 if [[ $RUN_ALL_TESTS = "1" ]]; then
   GO_TEST_TARGET="./..."
-  GO_TEST_MODULES=$(find . -name go.mod)
-  echo "Running all tests"
+  GO_TEST_MODULES=""
+  set +x
+  echo "Running all tests compatible with current go version"
+  # Test with all go modules compatible with go version under test.
+  for i in $(find . -name go.mod); do
+    mod="$(dirname $i)"
+    echo "Running 'version compatibility check' for '$mod'"
+    go run $ROOT_DIR/testing/kokoro/version.go -module=$mod && GO_TEST_MODULES="$GO_TEST_MODULES $mod"
+  done
+  set -x
 elif [[ -z "${TARGET_DIRS// }" ]]; then
```

Comment:
I guess the root is always compatible with all versions?

Reply:
My assumption is that the root module will always be the longest tail. It is checked where the root module is part of a list of all/changed modules.

FWIW, I don't understand the purpose of the root tests.

Reply:
The root tests do stuff like region tag checks and license header checks. So, we should always run them, regardless of the PR. I think it's safe to assume the oldest Go version will work with them.

Reply:
Ah, and those tests traverse the entire repo so only need to be run from root.